util/topgen/intermodule.py

```diff
@@ -69,6 +69,93 @@ def get_suffixes(ims: OrderedDict) -> (str, str):
     return ("_req", "_rsp")
 
 
+def add_intermodule_connection(obj, req_m, req_s, rsp_m, rsp_s):
+    """Add if doesn't exist the connection
+    """
+    req_key = "{}.{}".format(req_m, req_s)
+    rsp_key = "{}.{}".format(rsp_m, rsp_s)
+
+    connect = obj["inter_module"]["connect"]
+    if req_key in connect:
+        # check if rsp has data
+        if rsp_key in connect[req_key]:
+            return
+        req_key.append(rsp_key)
+        return
+
+    # req_key is not in connect:
+    # check if rsp_key
+    if rsp_key in connect:
+        # check if rsp has data
+        if req_key in connect[rsp_key]:
+            return
+        rsp_key.append(req_key)
+        return
+
+    # Add new key and connect
+    connect[req_key] = [rsp_key]
+
+
+def autoconnect(topcfg: OrderedDict):
+    """Matching the connection based on the naming rule
+    between {memory, module} <-> Xbar.
+    """
+
+    instances = topcfg["module"] + topcfg["memory"]
```

Comment:
this is not for this PR.  Do you think we should create "shell" hjson entries for things like ibex, rv_dm etc? Then maybe long term we can hook those into auto-connect also.

Reply:
yeah. that is always on my todo list.. :)